packages/graphiql-react/src/translation/services/translation-service.ts

```diff
@@ -0,0 +1,57 @@
+import { Language } from '../models/language';
+import { TranslationBundle } from '../models/translation-bundle';
+
+export class TranslationService {
+  private readonly translationBundles: Record<string, TranslationBundle> = {};
+
+  constructor(translationBundles: Record<string, TranslationBundle>) {
+    this.translationBundles = translationBundles;
+  }
+
+  translate(
+    key: string,
+    language: Language,
+    params: Record<string, string> = {},
+  ): string {
+    const translationBundle = this.translationBundles[language.code];
+    let translation = undefined;
+    if (translationBundle) {
+      translation = translationBundle.getTranslation(key);
+    }
+
+    translation ||=
+      this.translationBundles[Language.EN.code].getTranslation(key);
+
+    if (translation) {
+      return this.applyParameters(translation, params);
+    }
+    return key;
+  }
+
+  private applyParameters(
+    translation: string,
+    parameters: Record<string, string> = {},
+  ): string {
+    if (!parameters || Object.keys(parameters).length === 0) {
+      return translation;
+    }
+
+    let filledTranslation = '';
+    for (const key in parameters) {
+      filledTranslation = TranslationService.replaceAll(
+        translation,
+        key,
+        parameters[key],
+      );
+    }
+    return filledTranslation;
+  }
+
+  private static replaceAll(
```

Comment:
This method is quite artificial IMO. Also, string.replaceAll is better to be used instead of split/join

Reply:
Done